pkg/entity/batch_job/batch_job.go

```diff
@@ -69,25 +70,34 @@ func create(cmd *cobra.Command) {
 	if err != nil {
 		common.CliExit(err)
 	}
-	setCommonProjectIds(batchJob)
+	projectName := util.GetStringAndErr(flags, projectName)
+
+	var projectId string
+	if len(projectName) > 0 {
+		projectId = project.GetProjectId(projectName)
+	} else {
+		projectId = common.ProjectId
+	}
+
+	setCommonProjectIds(batchJob, projectId)
 	createBatchJobRequest := &batch_jobs.CreateBatchJobRequest{BatchJob: batchJob}
 	response, err := client.CreateBatchJob(apiContext, createBatchJobRequest)
 	common.CliExit(err)
 
 	printer.Print(response)
 }
 
-func setCommonProjectIds(batchJob *entities.BatchJob) {
+func setCommonProjectIds(batchJob *entities.BatchJob, projectId string) {
```

Comment:
Unrelated to this PR, but question on this: we assume that all data connectors and the batch job itself reside in the same project, because we don't isolate projects properly enough, right?

Reply:
correct afaik